src/gpodder/plugins/itunes.py

```diff
@@ -69,6 +85,39 @@ def __init__(self):
         self.priority = directory.Provider.PRIORITY_SECONDARY_SEARCH
 
     def on_search(self, query):
-        json_url = 'https://itunes.apple.com/search?media=podcast&term={}'.format(urllib.parse.quote(query))
-
-        return [directory.DirectoryEntry(entry['collectionName'], entry['feedUrl'], entry['artworkUrl100']) for entry in util.read_json(json_url)['results']]
+        offset = 0
+
+        while True:
+            json_url = (
+                f'https://itunes.apple.com/search?media=podcast&term='
+                f'{urllib.parse.quote(query)}&limit={PAGE_SIZE}&offset='
+                f'{offset}'
+            )
+            json_data = util.read_json(json_url)
+
+            if json_data['resultCount'] > 0:
+                for entry in json_data['results']:
+                    if 'feedUrl' not in entry:
+                        continue
+
+                    title = entry['collectionName']
+                    url = entry['feedUrl']
+                    image = entry['artworkUrl100']
+
+                    yield directory.DirectoryEntry(title, url, image)
+                    returned_res += 1
+
+                offset += json_data['resultCount']
+            else:
+            # Unlike the podverse stop condition where we detect a resultCount
+            # smaller than the page size for apple we can only stop when 0
+            # results are returned because the API seems to consistently
+            # return more than the page size and does this in an inconsistent
+            # fasion, most often returning 210 results but based on my
+            # observartion any number between page size and page size + 10 is
+            # possible.
+            #
+            # With an API that does not obey its own rules the only valid stop
+            # condition is no results.
+
+                break
```

Comment:
Could be simplified to:

```suggestion
            if json_data['resultCount'] == 0:
                return
                
            for entry in json_data['results']:
                if 'feedUrl' not in entry:
                    continue

                title = entry['collectionName']
                url = entry['feedUrl']
                image = entry['artworkUrl100']

                yield directory.DirectoryEntry(title, url, image)
                returned_res += 1

            offset += json_data['resultCount']
```